### PR TITLE
DMP-1434: Support validation around the upload  audio flow

### PR DESCRIPTION
**Pull request by mestebanez**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-1434

### Change description ###

Adds unique validation codes for:-

* unprovided audio
* unsupported audio duration
* unsupported audio content type

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        // check the duration
+        OffsetDateTime startDate = addAudioMetadataRequest.getStartedAt();
+        OffsetDateTime finishDate = addAudioMetadataRequest.getEndedAt();
+
+        Duration difference = Duration.between(startDate, finishDate);
+        long minutesDifference = difference.get(ChronoUnit.SECONDS) / 60;
+
+        if (minutesDifference > audioDurationInMinutes) {
+            throw new DartsApiException(AudioApiError.FILE_DURATION_OUT_OF_BOUNDS);
+        }
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java:55]: A value of say `86401` seconds is logically greater than the configured max of `1440` minutes, but I think the integer division on L51 will knock it down to `1440`, meaning the exception will not be triggered. Is that the intention?

Suggest discarding the manual conversion in favour of using compareTo() on a `Duration` type (see prior comment in this class, and `application.yaml`.

File: src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    public void validate(AddAudioMetadataRequest addAudioMetadataRequest) {
+
+        // attempt to resolve the court house
+        retrieveCoreObjectService.retrieveCourthouse(addAudioMetadataRequest.getCourthouse());
+
+        log.debug("Validated the court house {} exists", addAudioMetadataRequest.getCourthouse());
+
+        // now check the white list for extension types
+        Optional<String> found = properties.getAllowedMediaFormats().stream().filter(ext -> addAudioMetadataRequest.getFormat().equals(ext)
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java]: Done

File: src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        ).findFirst();
+
+        if (!found.isPresent()) {
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java]: Done

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -105,6 +105,7 @@ darts:
   automated-tasks-pod: ${AUTOMATED_TASK_MODE:true}
   api-pod: ${API_MODE:true}
   audio:
+    max-file-duration-minutes: 1440
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/main/resources/application.yaml:108]: Done

File: src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+@RequiredArgsConstructor
+@Slf4j
+public class AddAudioValidator implements Validator<AddAudioMetadataRequest> {
+
+    private final RetrieveCoreObjectService retrieveCoreObjectService;
+
+    private final AudioConfigurationProperties properties;
+
+    @Value("${darts.audio.max-file-duration-minutes}")
+    private Long audioDurationInMinutes;
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java]: Done

File: src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java (hunk 1 above)
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/audio/validation/AddAudioValidator.java]: Nope this was an oversight. Duration use is much cleaner. Done

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**